antarest/study/business/output_variables_management.py

```diff
@@ -0,0 +1,174 @@
+# Copyright (c) 2025, RTE (https://www.rte-france.com)
+#
+# See AUTHORS.txt
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# SPDX-License-Identifier: MPL-2.0
+#
+# This file is part of the Antares project.
+from pathlib import Path
+from typing import Any, Iterator
+
+from antarest.study.business.aggregator_management import (
+    AggregatorManager,
+    MCAllAreasQueryFile,
+    MCAllLinksQueryFile,
+    MCIndAreasQueryFile,
+    MCIndLinksQueryFile,
+    MCRoot,
+    QueryFileType,
+)
+from antarest.study.storage.output_model import OutputVariablesMetadata
+from antarest.study.storage.rawstudy.model.filesystem.matrix.matrix import MatrixFrequency
+
+
+class OutputVariablesManager:
+    """
+    Manage output variables view.
+    """
+
+    @staticmethod
+    def _filter_files_with_same_prefix(
+        parent_folder: Path, file_type_class: type[QueryFileType]
+    ) -> dict[QueryFileType, MatrixFrequency]:
+        """
+        Iterate over all existing files inside the parent_folder.
+        Returns only one file for each query file class.
+        This way we avoid opening files we know will have the same headers
+
+        Example:
+            In the folder, there's 4 files:
+            `values-hourly.txt`, `values-weekly.txt`, `values-monthly.txt` and `details-annual.txt`
+            The method will only return 1 `values` file and the `details` one.
+
+        Args:
+            - parent_folder: parent folder containing several files
+            - file_type_class: query file type class.
+
+        Returns:
+            - A mapping from file type class to a certain existing matrix frequency
+        """
+        all_files = [d.name for d in parent_folder.iterdir()]
+        file_dict = {}
+        for file in all_files:
+            splitted_file_name = file.removesuffix(".txt").split("-")
+            if len(splitted_file_name) == 2:
+                file_type, freq = splitted_file_name
+            else:
+                file_type1, file_type2, freq = splitted_file_name
+                file_type = f"{file_type1}-{file_type2}"
+            file_dict[file_type_class(file_type)] = MatrixFrequency(freq)
+        return file_dict
+
+    @staticmethod
+    def _read_headers_only(
+        file_path: Path, mc_root: MCRoot, freq: MatrixFrequency, file_type: QueryFileType
+    ) -> dict[str, set[str]]:
```

Comment:
Wed'better have a dataclass for the return type, to clarify what it is, for example:
```python
@dataclass(frozen=True)
class ColumnHeader:
    name: str
    sub_columns_names: list[str]
```
and have a `list[ColumnHeader]` as the return type